pandas/core/arrays/datetimelike.py

```diff
@@ -459,7 +460,9 @@ def view(self: DatetimeLikeArrayT) -> DatetimeLikeArrayT:
         ...
 
     @overload
-    def view(self, dtype: Optional[Dtype] = ...) -> ArrayLike:
+    def view(
+        self: DatetimeLikeArrayT, dtype: Optional[Dtype] = ...
+    ) -> DatetimeLikeArrayT:
```

Comment:
hmm, what if dtype is say 'i8'. the return type won't be the same as self

Reply:
I saw you addressed that in a recent push to master